quesma/queryparser/pancake_aggregation_parser_buckets.go

```diff
@@ -382,3 +378,30 @@ func (cw *ClickhouseQueryTranslator) pancakeFindMetricAggregation(queryMap Query
 	}
 	return notFoundValue
 }
+
+// samplerRaw - in a proper request should be of QueryMap type.
+func (cw *ClickhouseQueryTranslator) parseSampler(samplerRaw any) bucket_aggregations.Sampler {
+	const defaultSize = 100
+	sampler, ok := samplerRaw.(QueryMap)
+	if !ok {
+		logger.WarnWithCtx(cw.Ctx).Msgf("sampler is not a map, but %T, value: %v", samplerRaw, samplerRaw)
+		return bucket_aggregations.NewSampler(cw.Ctx, defaultSize)
+	}
+	return bucket_aggregations.NewSampler(cw.Ctx, cw.parseIntField(sampler, "shard_size", defaultSize))
```

Comment:
I believe in previous logic we multiple original sampler as it was per shard few times.